### ISSUE TITLE
Allow read for all in egg-info file permissions

### ISSUE THREAD

**Issue by cdeil** [healpy/healpy#220]
I have `healpy` installed via Macports and am getting this error when I try to build the docs for `photutils` (cc @astrofrog, @larrybradley):
```
$ python2.7 setup.py build_sphinx
Traceback (most recent call last):
  File "setup.py", line 8, in <module>
    import ah_bootstrap
  File "/Users/deil/code/photutils/ah_bootstrap.py", line 85, in <module>
    use_setuptools()
  File "/Users/deil/code/photutils/ez_setup.py", line 130, in use_setuptools
    import pkg_resources
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2727, in <module>
    add_activation_listener(lambda dist: dist.activate())
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 700, in subscribe
    callback(dist)
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2727, in <lambda>
    add_activation_listener(lambda dist: dist.activate())
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2227, in activate
    self.insert_on(path)
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2334, in insert_on
    self.check_version_conflict()
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2373, in check_version_conflict
    for modname in self._get_metadata('top_level.txt'):
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2221, in _get_metadata
    for line in self.get_metadata_lines(name):
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 1209, in get_metadata_lines
    return yield_lines(self.get_metadata(name))
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 1201, in get_metadata
    return self._get(self._fn(self.egg_info,name))
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 1316, in _get
    stream = open(path, 'rb')
IOError: [Errno 13] Permission denied: '/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/healpy-1.8.2-py2.7.egg-info/top_level.txt'
```
I'm not sure if this is an issue in the `python setup.py build_sphinx` implementation in `photutils`, but in any case `healpy` should allow read for all when installing it's `egg-info`, no?
```
$ ls -hl /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/healpy-1.8.2-py2.7.egg-info/
total 56
-rw-r-----  1 root  wheel   939B Jan 15 22:23 PKG-INFO
-rw-r-----  1 root  wheel    15K Jan 15 22:23 SOURCES.txt
-rw-r-----  1 root  wheel     1B Jan 15 22:23 dependency_links.txt
-rw-r-----  1 root  wheel     7B Jan 15 22:23 top_level.txt
```

**Comment by lpsinger** [comment]
The problem is the permissions in the source tarball:

    $ tar -xzf healpy-1.8.2.tar.gz && ls -l healpy-1.8.2
    total 184
    -rw-r-----    1 lady3jane  wheel   2682 Jan 13 13:06 CHANGELOG.rst
    -rw-r-----    1 lady3jane  wheel   1171 Jan 13 13:00 CITATION
    -rw-r-----    1 lady3jane  wheel  18011 Jan 13 13:00 COPYING
    -rw-r-----    1 lady3jane  wheel   4521 Jan 13 13:00 INSTALL.rst
    -rw-r-----    1 lady3jane  wheel    586 Jan 13 13:00 MANIFEST.in
    -rw-r-----    1 lady3jane  wheel    939 Jan 13 13:14 PKG-INFO
    -rw-r-----    1 lady3jane  wheel   4815 Jan 13 13:00 README.rst
    drwxr-x---  131 lady3jane  wheel   4454 Jan 13 13:14 cfitsio
    -rw-r-----    1 lady3jane  wheel  10332 Jan 13 13:00 ez_setup.py
    drwxr-x---    3 lady3jane  wheel    102 Jan 13 13:14 healpixsubmodule
    drwxr-x---   17 lady3jane  wheel    578 Jan 13 13:14 healpy
    drwxr-x---    6 lady3jane  wheel    204 Jan 13 13:14 healpy.egg-info
    -rw-r-----    1 lady3jane  wheel    276 Jan 13 13:00 run_pykg_config.py
    -rw-r-----    1 lady3jane  wheel     59 Jan 13 13:14 setup.cfg
    -rwxr-x---    1 lady3jane  wheel  18258 Jan 13 13:00 setup.py

@zonca: on the machine where you created the tarball, is your umask set to a very restrictive value?

**Comment by zonca** [comment]
ok, try with `1.8.3`:

```bash
radish ~/p/software/healpy/dist/healpy-1.8.3 master$ pwd 
/home/zonca/p/software/healpy/dist/healpy-1.8.3
radish ~/p/software/healpy/dist/healpy-1.8.3 master$ ll
total 116
drwxrwxr-x 6 zonca zonca  4096 Jan 16 12:59 ./
drwxrwxr-x 4 zonca zonca  4096 Jan 16 13:00 ../
drwxrwxr-x 3 zonca zonca  4096 Jan 16 12:59 cfitsio/
-rw-rw-r-- 1 zonca zonca  2803 Jan 16 12:58 CHANGELOG.rst
-rw-rw-r-- 1 zonca zonca  1171 Jan 13 10:00 CITATION
-rw-rw-r-- 1 zonca zonca 18011 Jan 13 10:00 COPYING
-rw-rw-r-- 1 zonca zonca 10332 Jan 13 10:00 ez_setup.py
drwxrwxr-x 3 zonca zonca  4096 Jan 16 12:59 healpixsubmodule/
drwxrwxr-x 5 zonca zonca  4096 Jan 16 12:59 healpy/
drwxrwxr-x 2 zonca zonca  4096 Jan 16 12:59 healpy.egg-info/
-rw-rw-r-- 1 zonca zonca  4521 Jan 13 10:00 INSTALL.rst
-rw-rw-r-- 1 zonca zonca   586 Jan 13 10:00 MANIFEST.in
-rw-rw-r-- 1 zonca zonca   939 Jan 16 12:59 PKG-INFO
-rw-rw-r-- 1 zonca zonca  4815 Jan 13 10:00 README.rst
-rw-rw-r-- 1 zonca zonca   276 Jan 13 10:00 run_pykg_config.py
-rw-rw-r-- 1 zonca zonca    59 Jan 16 12:59 setup.cfg
-rw-rw-r-- 1 zonca zonca 18258 Jan 13 10:00 setup.py
```

**Comment by lpsinger** [comment]
Permissions look fine now:

    $ wget https://pypi.python.org/packages/source/h/healpy/healpy-1.8.3.tar.gz && tar -xzf healpy-1.8.3.tar.gz && ls -l healpy-1.8.3
    ...
    -rw-r--r--    1 lpsinger  wheel   2803 Jan 16 15:58 CHANGELOG.rst
    -rw-r--r--    1 lpsinger  wheel   1171 Jan 13 13:00 CITATION
    -rw-r--r--    1 lpsinger  wheel  18011 Jan 13 13:00 COPYING
    -rw-r--r--    1 lpsinger  wheel   4521 Jan 13 13:00 INSTALL.rst
    -rw-r--r--    1 lpsinger  wheel    586 Jan 13 13:00 MANIFEST.in
    -rw-r--r--    1 lpsinger  wheel    939 Jan 16 15:59 PKG-INFO
    -rw-r--r--    1 lpsinger  wheel   4815 Jan 13 13:00 README.rst
    drwxr-xr-x  131 lpsinger  wheel   4454 Jan 16 15:59 cfitsio
    -rw-r--r--    1 lpsinger  wheel  10332 Jan 13 13:00 ez_setup.py
    drwxr-xr-x    3 lpsinger  wheel    102 Jan 16 15:59 healpixsubmodule
    drwxr-xr-x   17 lpsinger  wheel    578 Jan 16 15:59 healpy
    drwxr-xr-x    6 lpsinger  wheel    204 Jan 16 15:59 healpy.egg-info
    -rw-r--r--    1 lpsinger  wheel    276 Jan 13 13:00 run_pykg_config.py
    -rw-r--r--    1 lpsinger  wheel     59 Jan 16 15:59 setup.cfg
    -rw-r--r--    1 lpsinger  wheel  18258 Jan 13 13:00 setup.py

@cdeil, did this work?

**Comment by cdeil** [comment]
I don't have time to fiddle with my Macports installation now ... it's just a warning and it should go away when `healpy` gets updated in Macports.

Thanks for the super-quick fix and new release!

**Comment by lpsinger** [comment]
I should be able to update the MacPorts package by Sunday.

**Comment by cdeil** [comment]
Actually this issue prevents me from installing `astropy` master ... I'll simply uninstall `py27-healpy` for now, but @lpsinger thanks for putting in a fix quickly!
```
$ python2.7 setup.py install --user
Traceback (most recent call last):
  File "setup.py", line 8, in <module>
    import ah_bootstrap
  File "/Users/deil/code/astropy/ah_bootstrap.py", line 85, in <module>
    use_setuptools()
  File "/Users/deil/code/astropy/ez_setup.py", line 130, in use_setuptools
    import pkg_resources
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2727, in <module>
    add_activation_listener(lambda dist: dist.activate())
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 700, in subscribe
    callback(dist)
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2727, in <lambda>
    add_activation_listener(lambda dist: dist.activate())
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2227, in activate
    self.insert_on(path)
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2334, in insert_on
    self.check_version_conflict()
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2373, in check_version_conflict
    for modname in self._get_metadata('top_level.txt'):
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 2221, in _get_metadata
    for line in self.get_metadata_lines(name):
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 1209, in get_metadata_lines
    return yield_lines(self.get_metadata(name))
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 1201, in get_metadata
    return self._get(self._fn(self.egg_info,name))
  File "/Users/deil/Library/Python/2.7/lib/python/site-packages/distribute-0.6.24-py2.7.egg/pkg_resources.py", line 1316, in _get
    stream = open(path, 'rb')
IOError: [Errno 13] Permission denied: '/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/healpy-1.8.2-py2.7.egg-info/top_level.txt'
```

**Comment by lpsinger** [comment]
> Actually this issue prevents me from installing astropy master ... I'll simply uninstall py27-healpy for now, but @lpsinger thanks for putting in a fix quickly!

Updated in MacPorts now.

**Comment by cdeil** [comment]
@lpsinger - Just to confirm ... the issue is resolved after I updated to healpy `1.8.4` in Macports. Thanks!
